### PR TITLE
Initial scripts for Icon Resumtion

### PR DESCRIPTION
**Pull request by HSavynetska**
Initial scripts for Icon Resumtion

### CODE REVIEW

**Comment by HSavynetska** [comment]
@dboltovskyi, @GetmanetsIrina, please review.

File: test_scripts/API/SetAppIcon/002_Mobile_app_registration_with_param_iconResumed_true.lua [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/API/SetAppIcon/commonIconResumed')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Variables ]]
+local requestParams = {
```
Comment (dboltovskyi) [inline test_scripts/API/SetAppIcon/002_Mobile_app_registration_with_param_iconResumed_true.lua]: @HSavynetska It's better not to use temporary variables ```requestParams``` and ```requestUiParams``` and to define the values directly in ```allParams```

File: test_scripts/API/SetAppIcon/003_Icon_resumption_by_disconnect.lua [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  requestParams = requestParams,
+  requestUiParams = requestUiParams
+}
+
+local function CloseConnection()
+	test.mobileConnection:Close()
+	EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered", { unexpectedDisconnect = true })
+end
+
+local function OpenConnection()
```
Comment (dboltovskyi) [inline test_scripts/API/SetAppIcon/003_Icon_resumption_by_disconnect.lua]: @HSavynetska It's better to move this function to common module

File: test_scripts/API/SetAppIcon/003_Icon_resumption_by_disconnect.lua [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    imageType = "DYNAMIC",
+    value = common.getPathToFileInStorage(requestParams.syncFileName)
+  }
+}
+local allParams = {
+  requestParams = requestParams,
+  requestUiParams = requestUiParams
+}
+
+local function CloseConnection()
```
Comment (dboltovskyi) [inline test_scripts/API/SetAppIcon/003_Icon_resumption_by_disconnect.lua]: @HSavynetska It's better to move this function to common module

File: test_scripts/API/SetAppIcon/003_Icon_resumption_by_disconnect.lua [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	test.mobileConnection:Close()
+	EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered", { unexpectedDisconnect = true })
```
Comment (dboltovskyi) [inline test_scripts/API/SetAppIcon/003_Icon_resumption_by_disconnect.lua]: @HSavynetska For better stability of the test expectation should precede the action

File: test_scripts/API/SetAppIcon/010_Mobile_App_received_response_SetAppIcon_REJECTED.lua [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+-- TBD
+--
+-- Description:
+-- In case:
+-- 1) SDL, HMI are started.
+-- 2) Mobile app is registered. Sends  PutFile and valid SetAppIcon requests.
+-- 3) HMI responds with REJECTED resultCode to SetAppIcon request. Mobile App received response SetAppIcon(REJECTED).
+-- 4) App is re-registered.
+-- SDL does:
+-- 1) Register an app successfully, respond to RAI with result code "SUCCESS", "iconResumed" = true.
```
Comment (dboltovskyi) [inline test_scripts/API/SetAppIcon/010_Mobile_App_received_response_SetAppIcon_REJECTED.lua]: @HSavynetska According to checklist: ```"iconResumed" = false.```

File: test_scripts/API/SetAppIcon/006_Two_apps_are_registered_App1_iconResumed_true_App2_iconResumed_true.lua [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/API/SetAppIcon/commonIconResumed')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Variables ]]
+local requestParams = {
+  syncFileName = "icon.png"
```
Comment (dboltovskyi) [inline test_scripts/API/SetAppIcon/006_Two_apps_are_registered_App1_iconResumed_true_App2_iconResumed_true.lua]: @HSavynetska Propose to set 2 different icons for the Apps in order to make sure that each app will be resumed with it's own icon

File: test_scripts/API/SetAppIcon/commonIconResumed.lua [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+            appName = config["application" .. pAppId].registerAppInterfaceParams.appName,
+            icon = pIconValue
+        }})
+      :Do(function(_, d1)
+          hmiAppIds[m.getConfigAppParams(pAppId).appID] = d1.params.application.appID
+        end)
+      :ValidIf(function(_,data)
+        if false == pIconResumed and
+          data.params.application.icon then
+          return false, "BC.OnAppRegistered notification contains unexpected icon value "
```
Comment (LuxoftAKutsan) [inline test_scripts/API/SetAppIcon/commonIconResumed.lua]: @HSavynetska please specify in error what icon value was expected and what was received

Comment (LuxoftAKutsan) [reply test_scripts/API/SetAppIcon/commonIconResumed.lua]: Please, rephrase : 
BC.OnAppRegistered notification contains unexpected parameter : icon

File: test_scripts/API/SetAppIcon/commonIconResumed.lua [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+function m.putFile(paramsSend, file, pAppId)
+  if paramsSend then
+    paramsSend = paramsSend
+  else paramsSend =  putFileAllParams()
+  end
+  if not pAppId then pAppId = 1 end
+  local mobSession = m.getMobileSession(pAppId)
+  local cid
+  if file ~= nil then
+    cid = mobSession:SendRPC("PutFile",paramsSend, file)
```
[the remark this answers is not in the document]
Comment (HSavynetska) [reply test_scripts/API/SetAppIcon/commonIconResumed.lua]: Done

File: test_scripts/API/SetAppIcon/commonIconResumed.lua [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+  if not pAppId then pAppId = 1 end
+  local mobSession = m.getMobileSession(pAppId)
+  local corId = mobSession:SendRPC("UnregisterAppInterface", { })
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered",
+    { appID = m.getHMIAppId(pAppId), unexpectedDisconnect = false })
+  mobSession:ExpectResponse(corId, { success = true, resultCode = "SUCCESS" })
+end
+
+--Description: Set all parameter for PutFile
+local function putFileAllParams()
```
[the remark this answers is not in the document]
Comment (HSavynetska) [reply test_scripts/API/SetAppIcon/commonIconResumed.lua]: Done

File: test_scripts/API/SetAppIcon/commonIconResumed.lua [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+--! @parameters:
+--! pAppId - application number (1, 2, etc.)
+--! pIconResumed - apps icon was resumed at system or is not resumed
+--! pReconnection - re-register mobile application
+--! @return: none
+--]]
+function m.registerAppWOPTU(pAppId, pIconResumed, pReconnection, pIconValue)
+  if not pAppId then pAppId = 1 end
+  if pIconResumed == true then
+    if not pIconValue then
```
[the remark this answers is not in the document]
Comment (HSavynetska) [reply test_scripts/API/SetAppIcon/commonIconResumed.lua]: ?

File: test_scripts/API/SetAppIcon/commonIconResumed.lua [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+local commonPreconditions = require('user_modules/shared_testcases/commonPreconditions')
+
+--[[ Module ]]
+local m = actions
+
+--[[ Variables ]]
+-- local ptuTable = {}
+local hmiAppIds = {}
+
+function m.getPathToFileInStorage(fileName, pAppId)
```
[the remark this answers is not in the document]
Comment (HSavynetska) [reply test_scripts/API/SetAppIcon/commonIconResumed.lua]: Done

File: test_scripts/API/SetAppIcon/commonIconResumed.lua [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+    RegisterApp()
+  else
+    mobSession:StartService(7)
+    :Do(function()
+      RegisterApp()
+    end)
+  end
+end
+
+--Description: unregisterAppInterface successfully
```
[the remark this answers is not in the document]
Comment (HSavynetska) [reply test_scripts/API/SetAppIcon/commonIconResumed.lua]: Done

File: test_scripts/API/SetAppIcon/commonIconResumed.lua [hunk 13]
```diff
@@ -0,0 +1,140 @@
...
+local actions = require("user_modules/sequences/actions")
+local utils = require("user_modules/utils")
+local test = require("user_modules/dummy_connecttest")
+local commonPreconditions = require('user_modules/shared_testcases/commonPreconditions')
+
+--[[ Module ]]
+local m = actions
+
+--[[ Variables ]]
+-- local ptuTable = {}
```
[the remark this answers is not in the document]
Comment (HSavynetska) [reply test_scripts/API/SetAppIcon/commonIconResumed.lua]: Done

File: test_scripts/API/SetAppIcon/003_Icon_resumption_by_disconnect.lua (hunk 2 above)
Comment (HSavynetska) [reply test_scripts/API/SetAppIcon/003_Icon_resumption_by_disconnect.lua]: Done

File: test_scripts/API/SetAppIcon/003_Icon_resumption_by_disconnect.lua (hunk 3 above)
Comment (HSavynetska) [reply test_scripts/API/SetAppIcon/003_Icon_resumption_by_disconnect.lua]: Done

File: test_scripts/API/SetAppIcon/006_Two_apps_are_registered_App1_iconResumed_true_App2_iconResumed_true.lua (hunk 6 above)
Comment (HSavynetska) [reply test_scripts/API/SetAppIcon/006_Two_apps_are_registered_App1_iconResumed_true_App2_iconResumed_true.lua]: Done

File: test_scripts/API/SetAppIcon/010_Mobile_App_received_response_SetAppIcon_REJECTED.lua (hunk 5 above)
Comment (GetmanetsIrina) [reply test_scripts/API/SetAppIcon/010_Mobile_App_received_response_SetAppIcon_REJECTED.lua]: case is updated, description is will be updated in checklist

Comment (GetmanetsIrina) [reply test_scripts/API/SetAppIcon/010_Mobile_App_received_response_SetAppIcon_REJECTED.lua]: case is updated, description will be updated in checklist.

**Pull request merged**